core/src/main/java/org/apache/carbondata/core/metadata/schema/table/DataMapSchema.java

```diff
@@ -290,4 +301,71 @@ public boolean isTimeSeries() {
   public void setTimeSeries(boolean timeSeries) {
     isTimeSeries = timeSeries;
   }
+
+  public boolean supportIncrementalBuild() {
+    String prop = getProperties().get(DataMapProperty.FULL_REFRESH);
+    return prop == null || prop.equalsIgnoreCase("false");
+  }
+
+  public String getPropertiesAsString() {
+    String[] properties = getProperties().entrySet().stream()
+        // ignore internal used property
+        .filter(p ->
+            !p.getKey().equalsIgnoreCase(DataMapProperty.DEFERRED_REBUILD) &&
+            !p.getKey().equalsIgnoreCase(DataMapProperty.CHILD_SELECT_QUERY) &&
```

Comment:
`DataMapProperty.CHILD_SELECT_QUERY` && `DataMapProperty.QUERY_TYPE`  was used for preaggregate. Please remove it.

Reply:
done